daemon/daemon_test.go

```diff
@@ -716,3 +717,78 @@ func (s *daemonSuite) TestRebootHelper(c *check.C) {
 		cmd.ForgetCalls()
 	}
 }
+
+func makeDaemonListeners(c *check.C, d *Daemon) {
+	snapdL, err := net.Listen("tcp", "127.0.0.1:0")
+	c.Assert(err, check.IsNil)
+
+	snapL, err := net.Listen("tcp", "127.0.0.1:0")
+	c.Assert(err, check.IsNil)
+
+	snapdAccept := make(chan struct{})
+	snapdClosed := make(chan struct{})
+	d.snapdListener = &witnessAcceptListener{Listener: snapdL, accept: snapdAccept, closed: snapdClosed}
+
+	snapAccept := make(chan struct{})
+	d.snapListener = &witnessAcceptListener{Listener: snapL, accept: snapAccept}
+}
+
+// This test tests that when the snapd calls a restart of the system
+// a sigterm (from e.g. systemd) is handled when it arrives before
+// stop is fully done.
+func (s *daemonSuite) TestRestartShutdownWithSigtermInBetween(c *check.C) {
+	oldRebootNoticeWait := rebootNoticeWait
+	defer func() {
+		rebootNoticeWait = oldRebootNoticeWait
+	}()
+	rebootNoticeWait = 150 * time.Millisecond
+
+	cmd := testutil.MockCommand(c, "shutdown", "")
+	defer cmd.Restore()
+
+	d := newTestDaemon(c)
+	makeDaemonListeners(c, d)
+	s.markSeeded(d)
+
+	d.Start()
+	d.overlord.State().RequestRestart(state.RestartSystem)
+
+	ch := make(chan os.Signal, 2)
+	go func() { ch <- syscall.SIGTERM }()
```

Comment:
not sure why the goroutine here?

Reply:
Ups, that was a leftover from a previous iteration, I removed it now.